setup.py

```diff
@@ -213,8 +229,62 @@ def check_linker_need_libatomic():
 CYTHON_HELPER_C_FILES = ()
 
 CORE_C_FILES = tuple(grpc_core_dependencies.CORE_SOURCE_FILES)
+if "linux" in sys.platform:
+    CORE_C_FILES = filter(lambda x: 'third_party/libuv/src/win' not in x,
+                          CORE_C_FILES)
+    # The following source files are under libuv/src/unix but only for Darwin.
+    # They need to be excluded when running on Linux, and must keep synchronous
+    # with darwin_uv_srcs in src/libuv/gen_build_yaml.py, except
+    # src/unix/proctitle.c which exists in both darwin_uv_srcs and
+    # linux_uv_srcs.
+    CORE_C_FILES = filter(
+        lambda x: 'third_party/libuv/src/unix/bsd-ifaddrs.c' not in x,
+        CORE_C_FILES)
+    CORE_C_FILES = filter(
+        lambda x: 'third_party/libuv/src/unix/darwin.c' not in x, CORE_C_FILES)
+    CORE_C_FILES = filter(
+        lambda x: 'third_party/libuv/src/unix/darwin-proctitle.c' not in x,
+        CORE_C_FILES)
+    CORE_C_FILES = filter(
+        lambda x: 'third_party/libuv/src/unix/fsevents.c' not in x,
+        CORE_C_FILES)
+    CORE_C_FILES = filter(
+        lambda x: 'third_party/libuv/src/unix/kqueue.c' not in x, CORE_C_FILES)
+    CORE_C_FILES = filter(
+        lambda x: 'third_party/libuv/src/unix/random-getentropy.c' not in x,
+        CORE_C_FILES)
```

Comment:
How about:
```python
ignore_core_c_files = []

if blah:
    ignore_core_c_files.append('third_party/libuv/src/unix/bsd-ifaddrs.c')
else:
    ignore_core_c_files.append(...)

CORE_C_FILES = filter(lambda x: x not in ignore_core_c_files, CORE_C_FILES)
```

These bunch of filter-plus-not-in statements can be merged into one list... It's quite inefficient from algorithm perspective.

Reply:
Thanks for the suggestion! Done.

Reply:
Can you also apply the same pattern for all file-excluding logic? You can find them through the `CORE_C_FILES = filter(lambda x:` prefix.

Reply:
Others may not work with this pattern since they are paths to directories but not files, so they are checking whether the string is part of a file path, e.g.

`CORE_C_FILES = filter(lambda x: 'third_party/cares' not in x, CORE_C_FILES)`